### ISSUE TITLE
Add an extension trait for Option as well

### ISSUE THREAD

**Issue by shepmaster** [shepmaster/snafu#22]
`Option::ok_or` / `Option::ok_or_else` are common ways of generating errors; we should support that.

Something like

```rust
#[derive(Debug, Snafu)]
enum Error {
    Missing,
}

let x: Option<i32> = None;
x.context(Missing)?;
```

I think this should only apply to types *without* a `source`. We may need to create our own `NoneError` to provide to the `Context` struct and will need to generate appropriate `From` implementations.

**Comment by mikeando** [comment]
I'm thinking this would be very useful. But I'd love it to support types where the source was optional.
I've got an Error type that can be caused by a missing field in a given structure, or an IO error. The higher level code doesn't really care which occurred. 

But maybe that can come as a later enhancement.

Hopefully, I can take a look at implementing this next week.

**Comment by shepmaster** [comment]
I also had a need for it and sketched out [a basic implementation](https://github.com/shepmaster/snafu/compare/optionext?expand=1). If you'd like, give that a shot and see if it meets (half of) your needs.

**Comment by shepmaster** [comment]
> I've got an Error type that can be caused by a missing field in a given structure, or an IO error.

This *feels* like it will be a separate addition, would you mind opening a new issue and we can hash out why such a feature is good and what this might look like when implemented?

**Comment by mikeando** [comment]
That implementation looks very much like what I had in mind. (Plus I like the From/Into change)

What do you see as still needing work / fixing for the OptionExt stuff? (I've not tried compiling / testing that yet... just eyeballed the changes).

**Comment by shepmaster** [comment]
> That implementation looks very much like what I had in mind.

That's enough for me!

**Comment by shepmaster** [comment]
Released in 0.1.6
